app/controllers/wpcc/your_results_controller.rb

```diff
@@ -31,10 +40,19 @@ def convert_salary_frequency(salary_frequency)
       Wpcc::SalaryFrequencyConverter.convert(salary_frequency)
     end
 
-    def present(your_results)
-      your_results.map do |period|
+    def present_period_contributions(your_results)
+      your_results.map do |period_contribution|
         Wpcc::PeriodContributionPresenter.new(
-          period,
+          period_contribution,
+          view_context: view_context
+        )
+      end
+    end
+
+    def present_period_contribution_calculations(period_percents)
+      period_percents.map do |period_percent|
+        Wpcc::PeriodContributionCalculatorPresenter.new(
+          period_percent,
           view_context: view_context
         )
       end
```

Comment:
I just give this suggestions because I am finding dificult to follow the code between public and private methods and between the business classes and the presenter classes along side with the params and the sessions.

Can I suggest something along the lines (This is a fake/brainstorm code): 

```ruby
module Wpcc
  class YourResultsController < EngineController
    def schedule
      calendar.schedule.map do |period_contribution|
        Wpcc::PeriodContributionPresenter.new(
          period_contribution,
          view_context: view_context
        )
      end
    end
    helper_method :schedule

    def period_percents
      calendar.period_percents.map do |period_percent|
        Wpcc::PeriodContributionCalculatorPresenter.new(
          period_percent,
          view_context: view_context
        )
      end
    end
    helper_method :period_percents

    def message_presenter
      Wpcc::MessagePresenter.new(
        salary_message,
        view_context: view_context
      )
    end
    helper_method :message_presenter

    private
    # ...
end
```